pumpkin/src/main.rs

```diff
@@ -25,6 +25,10 @@ use client::Client;
 use server::{ticker::Ticker, Server};
 use std::io::{self};
 use tokio::io::{AsyncBufReadExt, BufReader};
+#[cfg(windows)]
+use tokio::signal::ctrl_c;
```

Comment:
This is not windows specific

Reply:
oh true, this should be enabled on anything but unix. i'll try fix, thanks

Reply:
Resolved in 7ab44d0

Reply:
Oh man Please, I now tell you the third time. ctrlc can be included in **any** Platform. Tokio has interinal checking and uses the right platform. Your current implementation only sets ctrlc for only on Platform. Suprise ctrlc is supported on more platforms. Just take a look at https://github.com/tokio-rs/tokio/blob/master/tokio/src/signal/ctrl_c.rs

Reply:
I think I understand what you're requesting now, @Snowiiii: to alter the PR to just use ctrlc on all platforms. I advise against this,

Using ctrlc on UNIX has [caveats](https://github.com/tokio-rs/tokio/blob/master/tokio/src/signal/ctrl_c.rs#L18-L34), and thus for UNIX systems I am using a custom implementation without those caveats. Hope that explains why there are platform-dependent versions in this PR.

I'd be happy to alter the PR to just use ctrlc on all platforms and accept the caveats, but I'd personally recommend against it.

Reply:
Ohh so from my understanding it sends, A Singal when CTRL+C is pressed ?

Reply:
- Basically, UNIX has a whole different way of signal handling than Windows.
- The windows way of doing it afaik is purely ctrl-c requests the program to stop running.
- However, with UNIX there are a bunch more signals, and they are handled in a considerably different way to Windows.
- So although the `ctrlc` module in `tokio` 'works' cross-platform, since it tries to make everything work all in one, it can cause issues for UNIX systems in mostly edge-cases.
- This is why I have separated the UNIX signal handling from all other platforms.
- Since the UNIX signal handling code in this PR already handles the signals from UNIX's variant of 'Ctrl-C', there is no need to enable the `ctrlc` module code; if anything this may cause conflicts with the existing unix-specific signal handling code.

Hope that helps clear it up, let me know if you are still unsure about anything 🙂